File: mysql-async/src/main/scala/com/github/mauricio/async/db/mysql/binary/encoder/ByteBufferEncoder.scala
package com.github.mauricio.async.db.mysql.binary.encoder

import java.nio.ByteBuffer

import com.github.mauricio.async.db.mysql.column.ColumnTypes
import com.github.mauricio.async.db.util.ChannelWrapper.bufferToWrapper
import io.netty.buffer.ByteBuf

object ByteBufferEncoder extends BinaryEncoder {
  def encode(value: Any, buffer: ByteBuf) {
    val bytes = value.asInstanceOf[ByteBuffer]

    buffer.writeLength(bytes.remaining())
    buffer.writeBytes(bytes)
  }

  def encodesTo: Int = ColumnTypes.FIELD_TYPE_BLOB

}

File: mysql-async/src/main/java/com/github/jasync/sql/db/mysql/binary/encoder/ByteBufferEncoder.kt
package com.github.jasync.sql.db.mysql.binary.encoder

import com.github.jasync.sql.db.util.writeLength
import com.github.jasync.sql.db.mysql.column.ColumnTypes
import io.netty.buffer.ByteBuf
import java.nio.ByteBuffer

object ByteBufferEncoder : BinaryEncoder {
  override fun encode(value: Any, buffer: ByteBuf) {
    val bytes = value as ByteBuffer

    buffer.writeLength(bytes.remaining().toLong())
    buffer.writeBytes(bytes)
  }

  override fun encodesTo(): Int = ColumnTypes.FIELD_TYPE_BLOB

}
